Add Guide to Secure AWS HTTP APIs Using Asgardeo (#5253)
wso2/docs-is: en/asgardeo/docs/guides/api-security/aws/securing-aws-http-apis.md

---
template: templates/quick-start.html

> heading: Secure a Pharmacy User API with AWS API Gateway and Asgardeo

Reviewer: Shall we keep the heading generic?

```suggestion
heading: Securing APIs with AWS API Gateway and Asgardeo
```

Author: Addressed in [e4fc8aa](https://github.com/wso2/docs-is/pull/5253/commits/e4fc8aa8a7f3140b9f3447455ff995b2ad2ef2ac)

> description: This guide walks you through securing a Pharmacy User API using AWS API Gateway and WSO2 Asgardeo. You'll deploy a Lambda function, expose it via an HTTP API, and protect the endpoint using JWT validation powered by Asgardeo.

Reviewer: ```suggestion
description: This guide walks you through securing a Pharmacy User API using AWS API Gateway and Asgardeo. You'll deploy a Lambda function, expose it via an HTTP API, and protect the endpoint using JWT validation powered by Asgardeo.
```

JWT validation is independent of Asgardeo, isn't it? in that case, "JWT validation powered by Asgardeo." isn't accurate, IMO

Author: Addressed in [e4fc8aa](https://github.com/wso2/docs-is/pull/5253/commits/e4fc8aa8a7f3140b9f3447455ff995b2ad2ef2ac)

> what_you_will_learn:
>   - Deploy a serverless AWS Lambda endpoint
>   - Secure the API using AWS API Gateway's JWT authorizer
>   - Configure OAuth2 application and API scopes in Asgardeo
>   - Protect API access using JWTs and test secured endpoints

Reviewer: ```suggestion
what_you_will_learn:
  - Deploy a serverless endpoint using AWS Lambda
  - Secure your API with AWS API Gateway’s JWT authorizer
  - Configure an OAuth2 application and define API scopes in Asgardeo
  - Protect and test your API using JWT-based access control
```

prerequisites:
  - About 20 minutes
  - AWS Account with CLI access

>   - <a href="https://asgardeo.io" target="_blank" rel="noopener noreferrer">Asgardeo account</a>

Reviewer: ```suggestion
  - an <a href="https://asgardeo.io" target="_blank" rel="noopener noreferrer">Asgardeo account</a>
```

Author: Addressed in [e4fc8aa](https://github.com/wso2/docs-is/pull/5253/commits/e4fc8aa8a7f3140b9f3447455ff995b2ad2ef2ac)

  - Tools: <code>curl</code> and <code>jq</code>
  - Familiarity with shell and JSON
---

## Step 1: Create and Deploy the Lambda Function

### 1. Create IAM Role for Lambda Execution

> ```bash

Reviewer: Shall we add a description?

Author: Addressed in [e4fc8aa](https://github.com/wso2/docs-is/pull/5253/commits/e4fc8aa8a7f3140b9f3447455ff995b2ad2ef2ac)

  --api-id $API_ID \
  --integration-type AWS_PROXY \
  --integration-method POST \
  --integration-uri $FUNCTION_ARN \
  --payload-format-version 2.0 \
  --query 'IntegrationId' --output text)
echo "Created API integration with ID: $INTEGRATION_ID"
```

### 3. Create API Route
```bash
ROUTE_ID=$(aws apigatewayv2 create-route \
  --api-id $API_ID \
  --route-key "GET /user" \
  --target "integrations/$INTEGRATION_ID" \
  --authorization-type NONE \
  --query 'RouteId' --output text)
echo "Created route GET /user with ID: $ROUTE_ID"
```

### 4. Grant Permission to API Gateway
```bash
REGION=$(aws configure get region)
ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)

aws lambda add-permission \
  --function-name PharmacyUserFunction \
  --statement-id AllowInvokeByApiGateway \
  --action lambda:InvokeFunction \
  --principal apigateway.amazonaws.com \
  --source-arn "arn:aws:execute-api:$REGION:$ACCOUNT_ID:$API_ID/*/GET/user"
```

### 5. Deploy the API
```bash
aws apigatewayv2 create-stage --api-id $API_ID --stage-name dev --auto-deploy
echo "API deployed to stage 'dev'"
```

## Step 3: Obtain a Bearer Token for Asgardeo Manage App

> You will need a Client ID and Client Secret from an Asgardeo Manage application that can be used to call Asgardeo’s management APIs. You can use an existing application or create a new Manage application in the Asgardeo console with the Client Credentials grant enabled. Make sure the application has the scope internal_application_mgt_create (this scope is required to create a new application via the API).

Reviewer: ```suggestion
You will need a Client ID and Client Secret from an Asgardeo application that is authorized to invoke Asgardeo management APIs. You can use an existing application or create a new application in the Asgardeo console with the Client Credentials grant enabled. Make sure the application has the scope `internal_application_mgt_create` (this scope is required to create a new application via the API).
```

Author: Addressed in [e4fc8aa](https://github.com/wso2/docs-is/pull/5253/commits/e4fc8aa8a7f3140b9f3447455ff995b2ad2ef2ac)

> ```bash

Reviewer: ```suggestion

```bash
```

Author: Addressed in [e4fc8aa](https://github.com/wso2/docs-is/pull/5253/commits/e4fc8aa8a7f3140b9f3447455ff995b2ad2ef2ac)

curl -X POST "https://api.asgardeo.io/t/<YOUR_ASGARDEO_ORG_NAME>/oauth2/token" \
  -u "<YOUR_ADMIN_APP_CLIENT_ID>:<YOUR_ADMIN_APP_CLIENT_SECRET>" \
  -H "Content-Type: application/x-www-form-urlencoded" \
  -d "grant_type=client_credentials&scope=internal_application_mgt_create" 2>/dev/null | jq -r .access_token
```
Use the obtained token in all subsequent Asgardeo API calls:

> ```bash

Reviewer: ```suggestion

```bash
```

Author: Addressed in [e4fc8aa](https://github.com/wso2/docs-is/pull/5253/commits/e4fc8aa8a7f3140b9f3447455ff995b2ad2ef2ac)
